src/components/anime-search/AnimeSearch.tsx

```diff
@@ -0,0 +1,94 @@
+import {ChangeEvent, useState, useCallback, useRef} from "react";
+import { Icon } from "src/ui/icon/Icon";
+import { useModal } from "src/hooks/modal/Modal.tsx";
+import { Dialog } from "src/ui/dialog/dialog.tsx";
+import { Input } from "src/ui/input/input.tsx";
+import { useDebounce } from "src/hooks/debounce/Debounce.tsx";
+import { markText } from "src/utils/markText.tsx";
+import axios from "axios";
+import style from "./animeSeacrh.module.scss"
+import cnBind from 'classnames/bind'
+
+const cx = cnBind.bind(style)
+const url = (keyword: string) => `https://kinopoiskapiunofficial.tech/api/v2.2/films?keyword=${keyword}`
+const option = {
+  headers: {
+    'X-API-KEY': '4dc0fb6b-92e7-4e5d-b3c6-960b4ce6443d',
+    'Content-Type': 'application/json',
+  },
+}
+interface DataFilms {
+  kinopoiskId: number
+  nameRu: string
+  year: number
+}
+export const AnimeSearch = () => {
+  const [visible, changeVisible] = useModal();
+  const [dataFilms, setDataFilms] = useState([])
+  const [search, setSearch] = useState("")
+  const searchRef = useRef<string>(search)
+  searchRef.current = search
+
+  const debounce = useCallback(useDebounce(() => getDataFilms() , 2000), [])
+  const onClearValue = (): void => {
+    setSearch("" )
+  }
+  const getDataFilms = async () => {
+    try {
+      const data = await (await (axios.get(url(searchRef.current), option))).data.items.slice(0, 9).filter(({ nameRu }: DataFilms) => nameRu != null)
+      setDataFilms(data)
+    } catch (error) {
+      console.log(`${error} --- error`)
+    }
+  }
+  // DEBT: в дальнешем пересмотреть надобность функции onClearValue
+  const onValueChange = (event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    setSearch(event.target.value)
+    debounce()
+  }
+
+  return (
+      <>
+        <button className={cx("anime-search")} onClick={() => changeVisible()}>
```

Comment:
```suggestion
        <button className={cx("anime-search")} onClick={() => handleDialogVisibleToggle()}> иди что там у тебя
```

Все события, которые делает пользовать именуются через handle

Reply:
исправлено